libpyvinyl/Parameters/Parameter.py

```diff
@@ -1,34 +1,247 @@
+
+        :param name: name of the parameter
+        :type name: str
+
+        :param unit: physical units returning the parameter value
+        :type unit: str
+
+        :param comment: brief description of the parameter
+        :type comment: str
+
         """
         self.name = name
-        self.unit = unit
+        self.__unit = Unit(unit) if unit != None else ""
         self.comment = comment
-        self._value = None
-        self._intervals = []
-        self._intervals_are_legal = None
-        self._options = []
-        self._options_are_legal = None
+        self.__value = None
+        self.__intervals = []
+        self.__intervals_are_legal = None
+        self.__options = []
+        self.__options_are_legal = None
+        self.__value_type = None
 
     @classmethod
     def from_dict(cls, param_dict):
-        param = cls(param_dict["name"], param_dict["unit"], param_dict["comment"])
+        """
+        Helper class method creating a new object from a dictionary providing
+         - name: str MANDATORY
+         - unit: str
+         - comment: str
+         - ...
+        This class method is mainly used to allow dumping and loading the class from json
+        """
+        if not "name" in param_dict:
+            raise KeyError(
+                "name is a mandatory element of the dictionary, but has not been found"
+            )
+        param = cls(
+            param_dict["name"], param_dict["_Parameter__unit"], param_dict["comment"]
+        )
         for key in param_dict:
             param.__dict__[key] = param_dict[key]
+
+        # set the value type, making the necessary promotions
+        param.__set_value_type(param.value)
+        for interval in param.__intervals:
+            param.__set_value_type(interval[0])
+            param.__set_value_type(interval[1])
+        for option in param.__options:
+            param.__set_value_type(option)
         return param
 
+    @property
+    def unit(self) -> str:
+        """Returning the units as a string"""
+        return str(self.__unit)
+
+    @unit.setter
+    def unit(self, uni):
+        """
+        Assignement of the units
+
+        A pint.Unit is used if the string is recognized as a valid unit in the registry.
+        It is stored as a string otherwise.
+        """
+        try:
+            self.__unit = Unit(uni)
+        except pint.errors.UndefinedUnitError:
+            self.__unit = uni
+
+    @property
+    def value_no_conversion(self):
+        """
+        Returning the object stored in value with no conversions
+        """
+        return self.__value
+
+    @property
+    def value(self):
+        if isinstance(self.__value, Quantity):
+            return self.__value.m_as(self.__unit)
+        else:
+            return self.__value
+
+    @classmethod
+    # self is not needed...
+    def __is_type_compatible(self, t1: type, t2: Union[None, type]) -> bool:
```

Comment:
If `self` is not needed, .e.g. it doesn't need to interact or return the members of the class, I would suggest using `@staticmethod`:
```
@staticmethod
    def __is_type_compatible(t1: type, t2: Union[None, type]) -> bool:
```